Mathlib/Data/List/Basic.lean

```diff
@@ -1756,6 +1756,16 @@ theorem bind_congr {l : List α} {f g : α → List β} (h : ∀ x ∈ l, f x =
   (congr_arg List.join <| map_congr h : _)
 #align list.bind_congr List.bind_congr
 
+theorem bind_option_toList_of_eq_some {f : β → Option α} {g : β → α} {bs : List β}
+    (h : ∀ x ∈ bs, f x = some (g x)) : bs.bind (fun i => (f i).toList) = bs.map g := by
```

Comment:
This looks like it should be factored through `filterMap`: `bs.bind (fun i => (f i).toList) = bs.filterMap f`, and then the congruence of `filterMap` and equality to `map` should already exist